internal/datacoord/clustering_compaction_manager.go

```diff
@@ -224,252 +206,310 @@ func (t *ClusteringCompactionManager) gc() error {
 }
 
-func (t *ClusteringCompactionManager) checkAllJobState() error {
-	jobs := t.GetAllJobs()
-	for _, job := range jobs {
-		err := t.checkJobState(job)
-		if err != nil {
-			log.Error("fail to check job state", zap.Error(err))
-			job.state = failed
-			log.Warn("mark clustering compaction job failed", zap.Int64("tiggerID", job.triggerID), zap.Int64("collectionID", job.collectionID))
-			t.saveJob(job)
+func (t *ClusteringCompactionManager) processAllTasks() error {
+	triggers := t.meta.GetClusteringCompactionTasks()
+	for triggerID, tasks := range triggers {
+		for _, task := range tasks {
+			log.Debug("task state before process", zap.Int64("planID", task.GetPlanId()), zap.String("state", task.State.String()))
+			err := t.processTask(task)
+			if err != nil {
+				log.Error("fail in process task", zap.Int64("TriggerId", task.TriggerId), zap.Int64("collectionID", task.CollectionId), zap.Int64("planID", task.PlanId), zap.Error(err))
+				task.State = datapb.CompactionTaskState_failed
+			}
+			log.Debug("task state after process", zap.Int64("planID", task.GetPlanId()), zap.String("state", task.State.String()))
+			t.saveTask(task)
 		}
+		state, executingCnt, pipeliningCnt, completedCnt, failedCnt, timeoutCnt, initCnt, analyzingCnt, analyzedCnt, indexingCnt, indexedCnt := t.summaryCompactionTaskState(tasks)
+		log.Debug("compaction states", zap.Int64("triggerID", triggerID), zap.String("state", state.String()),
+			zap.Int("executingCnt", executingCnt),
+			zap.Int("pipeliningCnt", pipeliningCnt),
+			zap.Int("completedCnt", completedCnt),
+			zap.Int("failedCnt", failedCnt),
+			zap.Int("timeoutCnt", timeoutCnt),
+			zap.Int("initCnt", initCnt),
+			zap.Int("analyzingCnt", analyzingCnt),
+			zap.Int("analyzedCnt", analyzedCnt),
+			zap.Int("indexingCnt", indexingCnt),
+			zap.Int("indexedCnt", indexedCnt))
 	}
 	return nil
 }
 
-func (t *ClusteringCompactionManager) checkJobState(job *ClusteringCompactionJob) error {
-	if job.state == completed || job.state == failed || job.state == timeout {
+func (t *ClusteringCompactionManager) processTask(task *datapb.CompactionTask) error {
+	if task.State == datapb.CompactionTaskState_indexed || task.State == datapb.CompactionTaskState_failed || task.State == datapb.CompactionTaskState_timeout {
 		return nil
 	}
+	coll, err := t.handler.GetCollection(t.ctx, task.GetCollectionId())
+	if err != nil {
+		log.Warn("fail to get collection", zap.Int64("collectionID", task.GetCollectionId()), zap.Error(err))
+		return err
+	}
+	if coll == nil {
+		log.Warn("collection not found, it may be dropped, stop clustering compaction task", zap.Int64("collectionID", task.GetCollectionId()))
+		return merr.WrapErrCollectionNotFound(task.GetCollectionId())
+	}
 
-	if job.state == pipelining {
-		err := t.runCompactionJob(job)
+	switch task.State {
+	case datapb.CompactionTaskState_init:
+		return t.processInitTask(task)
+	case datapb.CompactionTaskState_compact_pipelining, datapb.CompactionTaskState_compact_executing:
```

Comment:
so what's the difference between pipelineing and executing?

Reply:
Actually there is no pipelining state in major compaction. This is just to keep consistent with old compactonState. It is compatible if we save all types of compaction in meta in the future.

Reply:
> so what's the difference between pipelineing and executing?

Actually there is no pipelining state in major compaction. This is just to keep consistent with old compactonState. It is compatible if we save all types of compaction in meta in the future.